### ISSUE TITLE
conda installation failed with status code '1'

### ISSUE THREAD

**Issue by cmulittlechild** [kevinrue/velociraptor#92]
![image](https://github.com/user-attachments/assets/62bc17f1-2940-4855-8978-d792e54d3bfc)

I'm a bit confused, has this been trying to reinstall? I am running on a Linux server and have micromamba and microcanda. How can I solve this error

Thanks

**Comment by kevinrue** [comment]
Am I understanding right that you are running velociraptor from within a conda environment?

I haven't done that in a while, but in that case, since velociraptor is indeed its own micromamba installation, I'm not sure how the system deals with a possible nesting of conda installations.

(I'm using 'conda' and related software interchangeably, I know we're talking about mamba and friends)

I guess I'd have to try and replicate your setup to see if I can reproduce but further information about your hardware environment (eg, laptop, HPC) would be helpful.

**Comment by LTLA** [comment]
FYI **basilisk** allows the user to define an external conda installation to avoid a double-install. Regardless, **velociraptor** will still build its own conda environment.

**Comment by cmulittlechild** [comment]
> Am I understanding right that you are running velociraptor from within a conda environment?
> 
> I haven't done that in a while, but in that case, since velociraptor is indeed its own micromamba installation, I'm not sure how the system deals with a possible nesting of conda installations.
> 
> (I'm using 'conda' and related software interchangeably, I know we're talking about mamba and friends)
> 
> I guess I'd have to try and replicate your setup to see if I can reproduce but further information about your hardware environment (eg, laptop, HPC) would be helpful.

Thanks for your reply!
I would give you my info about how i work.
Now i am working on the HPC, here is the version of the HPC.
![image](https://github.com/user-attachments/assets/4bb8bef6-94b6-4967-9ff7-7c932fd3c876)

And basicly, for safety, this HPC only supports singularity.
So, i am using a singularity image, like shortcake (on docker hub for single-cell), and the author of this image provides micromamba, and then i install the miniconda in this sif image by myself.
They also provide a singularity version, https://www.dropbox.com/scl/fo/lptb68dirr9wcncy77wsv/h?dl=0&e=2&rlkey=whhcaxuvxd1cz4fqoeyzy63bf, i am using shrotcake_full.3.0.0.sif.

when i follow your tutorial and run out <- scvelo(list(X=spliced, spliced=spliced, unspliced=unspliced))
it gives me an ERROR:Error in installConda(): conda installation failed with status code '1'.

So, basicly, i am using velociraptor within a singulairty environment. and i specify the micromamba env contained in the singularity.

if you need any other info, please ask me.

Thanks for your help!

**Comment by cmulittlechild** [comment]
> FYI **basilisk** allows the user to define an external conda installation to avoid a double-install. Regardless, **velociraptor** will still build its own conda environment.

So, you mean, even i specify a env has been installed, the basilisk would still run the installation programme to build his own one?

**Comment by kevinrue** [comment]
Exactly, that's how we ensured exact reproducibility. We make a new environment with a fully defined set of packages and versions. Any other approach would create all sorts of complications compromising reproducibility.

**Comment by cmulittlechild** [comment]
> Exactly, that's how we ensured exact reproducibility. We make a new environment with a fully defined set of packages and versions. Any other approach would create all sorts of complications compromising reproducibility.

ok, may be i understand now, i think i need to do is download the installation script of miniconda3 corresponding to the basilisk requirement in the cache folder manually, and then when i run the scvelo, the function would detect he script downloaded and run smoothly, am i on the right way?

**Comment by kevinrue** [comment]
I'm not sure I understand what you're planning to do but by all means feel free to try anything you can think of outside the documentation at your own risk. If it works feel free to let us know what you did and we can document it for others.

On our HPC where we also use singularity containers using 'apptainer', I only need a container with R installed. From there, install BiocManager and velociraptor (no conda needed) and let velociraptor do the rest. Makes sense?

**Comment by cmulittlechild** [comment]
> I'm not sure I understand what you're planning to do but by all means feel free to try anything you can think of outside the documentation at your own risk. If it works feel free to let us know what you did and we can document it for others.
> 
> On our HPC where we also use singularity containers using 'apptainer', I only need a container with R installed. From there, install BiocManager and velociraptor (no conda needed) and let velociraptor do the rest. Makes sense?

yes, very helpful! i would try and tell you the outcome, thanks very much!
